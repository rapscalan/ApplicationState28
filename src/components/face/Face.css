.Face {
  padding: 1rem;
  text-align: center;
  font-size: 50vh;
  margin: 1rem;
}
