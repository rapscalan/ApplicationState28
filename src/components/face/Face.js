import React from 'react';
import styles from './Face.css';
import { useSelector } from 'react-redux';
import { getFace } from '../../data/selectors/moodsSelectors';

const Face = () => {
  const face = useSelector(getFace);
  return <p className={styles.Face}>{face}</p>;
};

export default Face;
