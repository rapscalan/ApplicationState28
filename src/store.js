const { createStore } = require('redux');
import reducer from './data/reducers/moodsReducers';

export default createStore(
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
