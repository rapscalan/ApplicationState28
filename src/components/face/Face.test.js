import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../store'; 
import Face, {getFace} from './Face';

describe('Face component', () => {
  it('renders Face', () => {
    //const wrapper = shallow(<Face emoji="ðŸ˜€" />);
    const wrapper = mount(<Provider store={store}><Face emoji="ðŸ˜€" /></Provider>); 
    expect(wrapper).toMatchSnapshot();
  });
});
  
