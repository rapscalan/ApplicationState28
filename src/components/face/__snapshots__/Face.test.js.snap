// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Face component renders Face 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Face
    emoji="ðŸ˜€"
  >
    <p
      className="Face"
    >
      ðŸ¤¬
    </p>
  </Face>
</Provider>
`;
